paddle/phi/kernels/onednn/matmul_kernel.cc

```diff
@@ -152,6 +161,419 @@ void MatmulKernel(const Context &dev_ctx,
+    for (int i = 0; i < count; i++) {
+      if (scale_y_data[i] == 0.0)
+        output_shift_scale[i] = scale_out_data;
+      else
+        output_shift_scale[i] =
+            scale_out_data / (scale_x_data * scale_y_data[i]);
+    }
+    int mul_mask = is_multi_channel ? 1 : 0;
+    mul_attr.set_output_scales(mul_mask, output_shift_scale);
+
+    return mul_attr;
+  }
+
+  inner_product_forward CreateMulPrimitive(const memory &x_memory,
+                                           const memory &y_memory,
+                                           const memory::desc &dst_desc,
+                                           DenseTensor *output,
+                                           const OneDNNContext &dev_ctx) {
+    const auto x_desc = x_memory.get_desc();
+    const auto y_desc = y_memory.get_desc();
+    inner_product_forward::primitive_desc mul_prim_desc;
+
+    const auto &mul_desc = inner_product_forward::desc(
+        prop_kind::forward, x_desc, y_desc, dst_desc);
+
+    if (is_int8_) {
+      bool force_fp32_output =
+          dev_ctx.HasDnnAttr("force_fp32_output")
+              ? PADDLE_GET_CONST(bool, dev_ctx.GetDnnAttr("force_fp32_output"))
+              : false;
+      auto mul_attr = CreateMulAttr(dev_ctx, force_fp32_output);
+      mul_prim_desc =
+          inner_product_forward::primitive_desc(mul_desc, mul_attr, engine_);
+    } else {
+      mul_prim_desc = inner_product_forward::primitive_desc(mul_desc, engine_);
+    }
+
+    output_ = CreateDstMemory(mul_prim_desc, dev_ctx, output);
+
+    return inner_product_forward(mul_prim_desc);
+  }
+
+  void Execute() {
+    auto &astream = OneDNNContext::tls().get_stream();
+    (*mul_).execute(astream,
+                    {{DNNL_ARG_SRC, *x_input_},
+                     {DNNL_ARG_WEIGHTS, *y_input_},
+                     {DNNL_ARG_DST, *output_}});
+    astream.wait();
+  }
+
+  template <typename T>
+  DenseTensor UpdateDataFormat(const DenseTensor *data,
+                               int num_col_dims,
+                               const OneDNNContext &dev_ctx) {
+    DenseTensor x_tmp;
+    DenseTensor data_matrix;
+    // This code is enforcing plain (non-blocked) memory arrangement
+    // in order to flatten (reduce dimensionality) of DenseTensor later
+    auto src_mdesc = data->mem_desc();
+    auto dst_mdesc = data->dims().size() >= 4
+                         ? (data->dims().size() == 5
+                                ? CreateMemDescriptor<T>(
+                                      data, funcs::OneDNNMemoryFormat::ncdhw)
+                                : CreateMemDescriptor<T>(
+                                      data, funcs::OneDNNMemoryFormat::nchw))
+                         : src_mdesc;
+
+    if (src_mdesc != dst_mdesc) {
+      dev_ctx.template Alloc<T>(&x_tmp, data->memory_size());
+
+      Reorder(src_mdesc,
+              dst_mdesc,
+              phi::funcs::to_void_cast<T>(data->data<T>()),
```

Comment:
you can get rid of `phi::` here as well as in other places in the code where method is in phi namespace